### PR TITLE
Set node color based on page rank score

### PR DESCRIPTION
**Pull request by rkeulemans**
Solves the first checkbox of: #129

### CODE REVIEW

File: TriblerGUI/widgets/trustpage/js/drawing.js [hunk 1]
```diff
@@ -61,9 +61,30 @@ function drawNodes(svg, data, on_click) {
...
+    var nodeColor = d3.scaleLinear()
+        .domain([0, 0.5, 1])
+        .range(["red", "yellow", "green"]);
+
+    // Map relative to the minimum and maximum page rank of the graph
+    var rank_difference = data.max_page_rank - data.min_page_rank;
+    if (rank_difference === 0) {
+        return nodeColor(1);
+    }
+    console.log((node.page_rank - data.min_page_rank) / rank_difference);
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Remove debug logging

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ var state = {
     request_pending: false,
     x: width / 2,
     y: height / 2,
-    focus_pk: "self",
+    focus_pk: "10",
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/main.js]: I would guess this focus node is because of debugging purposes. Please change this.

**Comment by MaxVanDeursen** [comment]
retest this please

**Comment by MaxVanDeursen** [comment]
retest this please

**Comment by MaxVanDeursen** [comment]
@yinghaodai @ghabbenjansen Any idea why the Linux tests do not pass?

**Comment by MaxVanDeursen** [comment]
retest this please

File: TriblerGUI/widgets/trustpage/js/drawing.js [hunk 3]
```diff
@@ -61,9 +61,30 @@ function drawNodes(svg, data, on_click) {
...
+    // Map relative to the minimum and maximum page rank of the graph
+    var rank_difference = data.max_page_rank - data.min_page_rank;
+    if (rank_difference === 0) {
+        return nodeColor(1);
+    }
+
+    return nodeColor((node.page_rank - data.min_page_rank) / rank_difference);
 }
 
+
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Why two newlines between functions?

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 4]
```diff
@@ -91,18 +91,19 @@ function update(graph) {
...
     // List the neighbors in each node
     graph.nodes.forEach(function (node) {
-        node.neighbors = listNeighborsOf(graph.links, node.public_key).map(function (pk) {
-            return state.nodes[pk];
+        node.neighbors = listNeighborsOf(graph.links, node.public_key_string).map(function (pk) {
+            return state.nodes[graph.public_keys.indexOf(pk)];
         });
     });
 
+
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/main.js]: Why two newlines between functions?

File: TriblerGUI/widgets/trustpage/js/drawing.js [hunk 5]
```diff
@@ -35,7 +35,7 @@ function drawNodes(svg, data, on_click) {
     // Append a <circle> element to it.
     var circles = groups
         .append("circle")
-        .attr("fill", "#ff9b00")
+        .attr("fill", function(d) {return getNodeColor(d, data)})
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Why wouldn't you just directly use getNodeColor?

Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Not possible because we need to 'bind' d (i.e. the node) first.

**Comment by rkeulemans** [comment]
I do not understand why this is out-of-date with the master branch. Can someone fix this?

File: TriblerGUI/widgets/trustpage/js/drawing.js (hunk 5 above)
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: My bad, just noticed.

**Pull request merged**